Gems/Archive/Code/Source/Tools/ArchiveWriter.cpp

```diff
@@ -0,0 +1,263 @@
+
+    // ArchiveStreamDeleter for the GenericStream unique_ptr
+    ArchiveWriter::ArchiveStreamDeleter::ArchiveStreamDeleter() = default;
+    ArchiveWriter::ArchiveStreamDeleter::ArchiveStreamDeleter(bool shouldDelete)
+        : m_delete(shouldDelete)
+    {}
+
+    void ArchiveWriter::ArchiveStreamDeleter::operator()(AZ::IO::GenericStream* ptr) const
+    {
+        if (m_delete && ptr)
+        {
+            delete ptr;
+        }
+    }
+
+    ArchiveWriter::ArchiveWriter() = default;
+
+    ArchiveWriter::~ArchiveWriter()
+    {
+        UnmountArchive();
+    }
+
+    ArchiveWriter::ArchiveWriter(const ArchiveWriterSettings& writerSettings)
+        : m_archiveWriterSettings(writerSettings)
+    {}
+
+    ArchiveWriter::ArchiveWriter(AZ::IO::PathView archivePath, const ArchiveWriterSettings& writerSettings)
+        : m_archiveWriterSettings(writerSettings)
+    {
+        MountArchive(archivePath);
+    }
+
+    ArchiveWriter::ArchiveWriter(ArchiveStreamPtr archiveStream, const ArchiveWriterSettings& writerSettings)
+        : m_archiveWriterSettings(writerSettings)
+    {
+        MountArchive(AZStd::move(archiveStream));
+    }
+
+    bool ArchiveWriter::MountArchive(AZ::IO::PathView archivePath)
+    {
+        UnmountArchive();
+        AZ::IO::FixedMaxPath mountPath{ archivePath };
+        constexpr AZ::IO::OpenMode openMode =
+            AZ::IO::OpenMode::ModeCreatePath
+            | AZ::IO::OpenMode::ModeWrite;
+
+        m_archiveStream.reset(aznew AZ::IO::SystemFileStream(mountPath.c_str(), openMode));
+
+        return m_archiveStream->IsOpen();
+    }
+
+    bool ArchiveWriter::MountArchive(ArchiveStreamPtr archiveStream)
+    {
+        UnmountArchive();
+        m_archiveStream = AZStd::move(archiveStream);
+
+        return m_archiveStream != nullptr && m_archiveStream->IsOpen();
+    }
+
+    void ArchiveWriter::UnmountArchive()
+    {
+        if (m_archiveStream->IsOpen())
+        {
+            m_archiveStream->Close();
+        }
+    }
+
+    bool ArchiveWriter::Commit()
+    {
+        return true;
+    }
+
+    ArchiveAddToFileResult ArchiveWriter::AddFileToArchive(AZ::IO::GenericStream& inputStream,
+        const ArchiveWriterFileSettings& fileSettings)
+    {
+        AZStd::vector<AZStd::byte> fileData;
+        fileData.resize_no_construct(inputStream.GetLength());
+        AZ::IO::SizeType bytesRead = inputStream.Read(fileData.size(), fileData.data());
+        // Unable to read entire stream data into memory
+        if (bytesRead != fileData.size())
+        {
+            ArchiveAddToFileResult errorResult;
+            errorResult.m_relativeFilePath = fileSettings.m_relativeFilePath;
+            errorResult.m_compressionAlgorithm = fileSettings.m_compressionAlgorithm;
+            errorResult.m_resultOutcome = AZStd::unexpected(
+                ResultString::format(R"(Unable to successfully read all bytes(%zu) from input stream.)"
+                    " %llu bytes were read.",
```

Comment:
```suggestion
                    " %" PRIu64" bytes were read.",
```

Reply:
Actually %llu is correct here.
The Type of AZ::IO::SizeType is AZ::u64 which is always [unsigned long long](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/base.h#L182-L185)